pulp-glue/pulp_glue/common/openapi.py

```diff
@@ -53,36 +55,79 @@ class AuthProviderBase:
     Returned auth objects need to be compatible with `requests.auth.AuthBase`.
     """
 
-    def basic_auth(self) -> t.Optional[t.Union[t.Tuple[str, str], requests.auth.AuthBase]]:
```

Comment:
No, you don't get to change the interface.

Reply:
Yeah, but the way the `AuthProviderBase` is written the `__call__` method is incompatible with the original interface and also only calls the `basic_auth` method. This way every time we need to write a new AuthProvider we have to override the `__call__` method to: first, be compliant with `AuthBase` objects, and second: to call another method that would make more send for different authentication method.